Circular_kernel_3/include/CGAL/Circular_kernel_3/Circular_arc_3.h

```diff
@@ -73,8 +73,8 @@ namespace CGAL {
       : _full(false)
       {
         // l must pass through s and t, and s != t
-        CGAL_kernel_precondition(SK().has_on_3_object()(c,s));
-        CGAL_kernel_precondition(SK().has_on_3_object()(c,t));
+//        CGAL_kernel_precondition(SK().has_on_3_object()(c,s));
+//        CGAL_kernel_precondition(SK().has_on_3_object()(c,t));
```

Comment:
Not very happy with this resolution.

The circular kernel is riddled with assertions that have no chance to be verified if the kernel does not provide exact constructions and it will always break with triangulations_on_sphere_2 code which uses mostly EPICK. However, those are still interesting to have, but it would need something like `CGAL_exact_kernel_precondition(...)`...

Reply:
It could be something like:
```C++
CGAL_kernel_precondition(std::is_floating_point<FT>::value || SK().has_on_3_object()(c,s));
```
or a variation of the same idea: check the exactness of the kernel in the assertion.